c_src/filewatch_inotify.c

```diff
@@ -241,7 +270,7 @@ static void ready_input(ErlDrvData self_, ErlDrvEvent fd_)
             serialize_event(self, event);
         }
 
-        if (++self->n_events < MAX_EVENTS) {
+        if (self->n_events < MAX_EVENTS) {
```

Comment:
I have a feeling that something fishy is going on here. At a quick glance of this diff, I can't seem to find where `n_events` is incremented which makes the removal of the `++` suspicious.